src/scf/driver/scf_loop.cpp

```diff
@@ -142,18 +181,70 @@ MODULE_RUN(SCFLoop) {
             H_new.emplace_back(F_new.coefficient(i),
                                F_new.get_operator(i).clone());
 
-        // Step 5b: New electronic hamiltonian to new electronic energy
+        // Step 4b: New electronic hamiltonian to new electronic energy
         chemist::braket::BraKet H_00(new_psi, H_new, new_psi);
         auto e_new = egy_mod.run_as<elec_egy_pt<wf_type>>(H_00);
 
-        // Step 6: Converged?
-        // TODO: gradient and energy differences
+        // Step 5: New density
+        density_op_type rho_hat_new(new_psi.orbitals(), new_psi.occupations());
+        chemist::braket::BraKet P_mn_new(aos, rho_hat_new, aos);
+        const auto& P_new = density_mod.run_as<density_pt>(P_mn_new);
+        density_t rho_new(P_new, new_psi.orbitals());
 
+        // Step 6: Converged?
+        if(iter > 1) {
+            simde::type::tensor de;
+            de("") = e_new("") - e_old("");
+
+            // Orbital gradient: FPS-SPF
+            // TODO: module satisfying BraKet(aos, Commutator(F,P), aos)
+            chemist::braket::BraKet F_mn(aos, f_new, aos);
+            const auto& F_matrix = F_mod.run_as<fock_matrix_pt>(F_mn);
+            simde::type::tensor FP;
+            simde::type::tensor PF;
+            FP("m,l") = F_matrix("m,n") * P_new("n,l");
+            PF("m,l") = P_new("m,n") * F_matrix("n,l");
+
+            simde::type::tensor FPS;
+            simde::type::tensor SPF;
+            FPS("m,l") = FP("m,n") * S("n,l");
+            SPF("m,l") = S("m,n") * PF("n,l");
```

Comment:
```suggestion
            simde::type::tensor FPS;
            FPS("m,l") = F_matrix("m,n") * P_new("n,l");
            FPS("m,l") = FPS("m,n") * S("n,l");
            
            simde::type::tensor SPF;
            SPF("m,l") = P_new("m,n") * F_matrix("n,l");
            SPF("m,l") = S("m,n") * SPF("n,l");
```
This would cut down on intermediates.